internal/pkg/deploy/cloudformation/stack/env_integration_test.go

```diff
@@ -318,3 +314,127 @@ network:
+				var mft manifest.Environment
+				err := yaml.Unmarshal([]byte(rawMft), &mft)
+				require.NoError(t, err)
+				return &deploy.CreateEnvironmentInput{
+					Version: "1.x",
+					App: deploy.AppInformation{
+						AccountPrincipalARN: "arn:aws:iam::000000000:root",
+						Name:                "demo",
+					},
+					Name:                 "test",
+					CIDRPrefixListIDs:    []string{"pl-mockid"},
+					ArtifactBucketARN:    "arn:aws:s3:::mockbucket",
+					ArtifactBucketKeyARN: "arn:aws:kms:us-west-2:000000000:key/1234abcd-12ab-34cd-56ef-1234567890ab",
+					CustomResourcesURLs: map[string]string{
+						"CertificateValidationFunction": "https://mockbucket.s3-us-west-2.amazonaws.com/dns-cert-validator",
+						"DNSDelegationFunction":         "https://mockbucket.s3-us-west-2.amazonaws.com/dns-delegation",
+						"CustomDomainFunction":          "https://mockbucket.s3-us-west-2.amazonaws.com/custom-domain",
+						"UniqueJSONValuesFunction":      "https://mockbucket.s3-us-west-2.amazonaws.com/unique-json-values",
+					},
+					Mft:    &mft,
+					RawMft: []byte(rawMft),
+				}
+			}(),
+			wantedFileName: "template-with-old-and-new-http-fields.yml",
+		},
+		"generate template with embedded manifest file with the new http manifest fields": {
+			input: func() *deploy.CreateEnvironmentInput {
+				rawMft := `name: test
+type: Environment
+# Create the public ALB with certificates attached.
+cdn:
+  certificate: viewer-cert
+http:
+  public:
+    ingress:
+      cdn: true
+    access_logs:
+      bucket_name: accesslogsbucket
+      prefix: accesslogsbucketprefix
+    certificates:
+      - cert-1
+      - cert-2
+  private:
+    ingress:
+      vpc: true
+observability:
+  container_insights: true # Enable container insights.`
+				var mft manifest.Environment
+				err := yaml.Unmarshal([]byte(rawMft), &mft)
+				require.NoError(t, err)
+				return &deploy.CreateEnvironmentInput{
+					Version: "1.x",
+					App: deploy.AppInformation{
+						AccountPrincipalARN: "arn:aws:iam::000000000:root",
+						Name:                "demo",
+					},
+					Name:                 "test",
+					CIDRPrefixListIDs:    []string{"pl-mockid"},
+					ArtifactBucketARN:    "arn:aws:s3:::mockbucket",
+					ArtifactBucketKeyARN: "arn:aws:kms:us-west-2:000000000:key/1234abcd-12ab-34cd-56ef-1234567890ab",
+					CustomResourcesURLs: map[string]string{
+						"CertificateValidationFunction": "https://mockbucket.s3-us-west-2.amazonaws.com/dns-cert-validator",
+						"DNSDelegationFunction":         "https://mockbucket.s3-us-west-2.amazonaws.com/dns-delegation",
+						"CustomDomainFunction":          "https://mockbucket.s3-us-west-2.amazonaws.com/custom-domain",
+						"UniqueJSONValuesFunction":      "https://mockbucket.s3-us-west-2.amazonaws.com/unique-json-values",
+					},
+					Mft:    &mft,
+					RawMft: []byte(rawMft),
+				}
+			}(),
+			wantedFileName: "template-with-old-and-new-http-fields.yml",
+		},
+	}
+	for name, tc := range testCases {
```

Comment:
yayy thank you for the tests! I have one nit thought 💭 

I can see how this test suite can be used to ensure that the old/new manifest produce the same CFN template, effectively speaking. My nit thought is more from a semantic perspective - that these test does not convey the purpose that "we want to ensure the CFN template produced by the old manifest is the same as the new manifest". 

Imagine a new contributor coming to the codebase and see these tests, it wouldn't be obvious to them that *these two tests are meant to be viewed together*, and that their wanted template *has to point to the same template in order for them to serve their purpose*. 

Not knowing this, the contributor may modify one of these test cases (e.g. adding a new manifest field, changing the wanted template to a different one). The purpose of these tests will be lost.

I have a code snippet that shows an alternative way to structure the test, hopefully that will better convey the intention of the test:

```go
func TestEnvStack_Regression(t *testing.T) {
	t.Run("produce the same template after migrating environment manifest to the new field hierarchy", func(t *testing.T) {
		originalManifest := `name: test
type: Environment
http:
  public:
    security_groups:
      ingress:
        restrict_to:
          cdn: true
  private:
    security_groups:
      ingress:
        from_vpc: true`
		// NOTE: in newManifest, `http.public.security_groups.ingress.restrict_to.cdn` is migrated to `http.public.ingress.cdn`.
		// and `http.private.security_groups.ingress.from_vpc` is migrated to `http.private.ingress.vpc`.
		newManifest := `name: test
type: Environment
http:
  public:
    ingress:
      cdn: true
  private:
    ingress:
      vpc: true`

		// Get the template content given the original manifest.
		originalStack := stack.NewEnvStackConfig(&deploy.CreateEnvironmentInput{
			RawMft: []byte(originalManifest),
			// ...
		})
		originalTmpl, err := originalStack.Template()
		require.NoError(t, err, "should serialize the template given the original environment manifest")
		originalObj := make(map[any]any)
		require.NoError(t, yaml.Unmarshal([]byte(originalTmpl), originalObj))

		// Get the template content given the manifest after migrating to the new UX.
		newEnvStack := stack.NewEnvStackConfig(&deploy.CreateEnvironmentInput{
			RawMft: []byte(newManifest),
			// ...
		})
		newEnvStackTemplate, err := newEnvStack.Template()
		require.NoError(t, err, "should serialize the template given a migrated environment manifest")
		newEnvTemplateContent := make(map[any]any)
		require.NoError(t, yaml.Unmarshal([]byte(newEnvStackTemplate), newEnvTemplateContent))

		// Check that the template content does not change after migrating to new manifest hierarchy.
		delete(originalObj, "Metadata")
		delete(newEnvTemplateContent, "Metadata")
		require.Equal(t, originalObj, newEnvTemplateContent)

	})
}
```

Reply:
Sounds like a good idea to me.